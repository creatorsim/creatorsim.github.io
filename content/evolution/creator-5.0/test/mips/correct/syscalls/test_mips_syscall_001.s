#
# Creator (https://creatorsim.github.io/creator/)
#

.data

.align 2

  byte:			.byte 120,1
  half:			.half 34
  word:			.word -5678
  string:  	.ascii "This is another string"

.text
main:
	# print int
	li $v0, 1
	li $a0, 6673
	syscall

	# print byte
  lb $a0, byte
  syscall
  
  # print out of range byte
  li $t0, 4
  la $t1, byte
  lb $a0, 1($t1)
  syscall
  
  # print half
  lh $a0, half
  syscall
  
  # print negative int
  lw $a0, word
  syscall
  
  # print int
  li $a0, 1
  syscall
  
  # string string
  lw $a0, string
  syscall
