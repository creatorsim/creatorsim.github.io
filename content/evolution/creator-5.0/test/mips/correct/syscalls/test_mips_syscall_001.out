
6673
120
1
34
-5678
1
1416128883
cr[PC]:0x6c; ir[1,at]:0x200008; ir[2,v0]:0x1; ir[4,a0]:0x54686973; ir[8,t0]:0x4; ir[9,t1]:0x200000; keyboard[0x0]:''; display[0x0]:'6673120134-567811416128883'; 

